content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1066 @@
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the route handler is implemented as follows:
+
+```js file=src/controllers/post.js
+const getPostById = async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const post = await prisma.post.findUnique({
+      where: { id },
+      include: {
+        author: true,
+        category: true,
+      }
+    })
+
+    if (!post) return res.status(404).json({ message: 'Post not found' })
+
+    return res.status(200).json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+### 4.2. Replacing queries in `POST` requests
+
+The REST API has three routes that accept `POST` requests:
+
+- `/user`: Creates a new `User` record
+- `/post`: Creates a new `User` record
+- `/user/:id/profile`: Creates a new `Profile` record for a `User` record with a given ID
+
+#### `/user`
+
+The `/user` handler is currently implemented as follows:
+
+```js file=src/controllers/user.js
+const createUser = async (req, res) => {
+  const { name, email } = req.body
+
+  try {
+    const user = await User.create({
+      name,
+      email,
+    })
+
+    return res.status(201).json(user)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the route handler is implemented as follows:
+
+```js file=src/controllers/user.js
+const createUser = async (req, res) => {
+  const { name, email } = req.body
+
+  try {
+    const user = await prisma.user.create({
+      data: {
+        name,
+        email,
+      },
+    })
+
+    return res.status(201).json(user)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+#### `/post`
+
+The `/post` handler is currently implemented as follows:
+
+```js file=src/controllers/post.js
+const createDraft = async (req, res) => {
+  const { title, content, authorEmail } = req.body
+
+  try {
+    const author = await User.findOne({ email: authorEmail })
+
+    if (!author) return res.status(404).json({ message: 'Author not found' })
+
+    const draft = await Post.create({
+      title,
+      content,
+      author: author._id,
+    })
+
+    res.status(201).json(draft)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the route handler is implemented as follows:
+
+```js file=src/controllers/post.js
+const createDraft = async (req, res) => {
+  const { title, content, authorEmail } = req.body
+
+  try {
+    const draft = await prisma.post.create({
+      data: {
+        title,
+        content,
+        author: {
+          connect: {
+            email: authorEmail,
+          },
+        },
+      },
+    })
+
+    res.status(201).json(draft)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+Note that Prisma Client's nested write here saves the initial query where the `User` record is first retrieved by its `email`. That's because, with Prisma you can connect records in relations using any unique property.
```

Comment:
Hmmmm this sentence is a little confusing to me. I don't see a `User` record above, only an `author`.

Also, I am not sure what you mean by "saves the initial query". Can you ping me and we can chat about what this means and then reword?

Reply:
> Also, I am not sure what you mean by "saves the initial query". Can you ping me and we can chat about what this means and then reword?

In the code snippet using Mongoose, there are 2 queries executed – the first the check if the user exists and the second to create a post record if the author/ user exists.

>  I don't see a User record above, only an `author`.

`author` is the name of the relation, which is of a`User` type.